src/kernels/ipywidgets-message-coordination/ipyWidgetScriptSourceProvider.ts

```diff
@@ -163,35 +160,12 @@ export class IPyWidgetScriptSourceProvider implements IWidgetScriptSourceProvide
     private rebuildProviders() {
         this.disposeScriptProviders();
 
-        const scriptProviders: IWidgetScriptSourceProvider[] = [];
-
-        // If we're allowed to use CDN providers, then use them, and use in order of preference.
-        if (this.configuredScriptSources.length > 0) {
-            scriptProviders.push(
-                new CDNWidgetScriptSourceProvider(this.configurationSettings, this.localResourceUriConverter, this.fs)
-            );
-        }
-        switch (this.kernel.kernelConnectionMetadata.kind) {
-            case 'connectToLiveRemoteKernel':
-            case 'startUsingRemoteKernelSpec':
-                scriptProviders.push(
-                    new RemoteWidgetScriptSourceProvider(this.kernel.kernelConnectionMetadata.baseUrl, this.httpClient)
-                );
-                break;
-
-            default:
-                scriptProviders.push(
-                    new LocalWidgetScriptSourceProvider(
-                        this.kernel,
-                        this.localResourceUriConverter,
-                        this.fs,
-                        this.interpreterService,
-                        this.factory
-                    )
-                );
-        }
-
-        this.scriptProviders = scriptProviders;
+        // Use the platform specific factory to get our providers
```

Comment:
In order to make this class platform agnostic, the creation of the script providers was moved to a platform specific factory.

The web version only support remote right now, but we might be able to do something about loading it from the CDN later.